MsixCore/msixmgr/msixmgr.cpp

```diff
@@ -411,25 +661,46 @@ int main(int argc, char * argv[])
                 {
                     if (!(EndsWith(unpackDestination, L".vhd") || (EndsWith(unpackDestination, L".vhdx"))))
                     {
+                        errorCode = msixmgr::GetErrorCodeFromHRESULT(E_INVALIDARG);
+                        errorDesc = L"Invalid VHD file name. File name must have .vhd or .vhdx file extension. HRESULT " + errorCode + L".";
+
                         std::wcout << std::endl;
                         std::wcout << "Invalid VHD file name. File name must have .vhd or .vhdx file extension" << std::endl;
                         std::wcout << std::endl;
+
+                        // Telemetry : Workflow Log
+                        QueryPerformanceCounter(&msixMgrLoad_EndCounter);
+                        workflowElapsedTime = msixmgr::CalcWorkflowElapsedTime(msixMgrLoad_StartCounter, msixMgrLoad_EndCounter, msixMgrLoad_Frequency);
+                        msixmgr::TraceLogWorkflow(workflowId.c_str(), cli.GetOperationTypeAsString().c_str(), false, workflowElapsedTime, errorCode.c_str(), errorDesc.c_str());
+
                         return E_INVALIDARG;
                     }
                     else
                     {
                         if (cli.GetVHDSize() == 0)

```

Comment:
Please take latest from master as VHDsize changes are merged and this section will have to be adjusted.

Reply:
Done